### PR TITLE
fix: insights event timeline perf and accuracy

### PR DESCRIPTION
**Pull request by alishaz-polymath**
## What does this PR do?

Fixes:
- Event Timeline Performance
- Event Timeline Accuracy
- Event Timeline range and URL param range discrepancy due to timezone
- Average duration Performance
- Average Duration Accuracy
- Most booked Members
- Least booked Members

- Fixes #XXXX (GitHub issue number)
- Fixes CAL-4312 (Linear issue number - should be visible at the bottom of the GitHub issue description)

## Mandatory Tasks (DO NOT REMOVE)

- [ ] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

File: packages/features/insights/server/events.ts [hunk 1]
```diff
@@ -1,13 +1,160 @@
...
+  ): Promise<AggregateResult> => {
+    // Determine the date truncation and date range based on timeView
+
+    const formattedStartDate = dayjs(startDate).format("YYYY-MM-DD HH:mm:ss");
+    const formattedEndDate = dayjs(endDate).format("YYYY-MM-DD HH:mm:ss");
+    const whereClause = buildSqlCondition(whereConditional);
+    const data = await prisma.$queryRaw<
+      { periodStart: Date; bookingsCount: number; timeStatus: string; noShowHost: boolean }[]
+    >`
+      SELECT
```
Comment (keithwillcode) [inline packages/features/insights/server/events.ts:84]: I'm testing this on prod now to see how it performs.

Comment (keithwillcode) [reply packages/features/insights/server/events.ts:84]: The resulting query runs in 13ms on prod. Amazing.

File: packages/features/insights/components/AverageEventDurationChart.tsx [hunk 2]
```diff
@@ -18,8 +19,8 @@ export const AverageEventDurationChart = () => {
   const initialConfigIsReady = !!(initialConfig?.teamId || initialConfig?.userId || initialConfig?.isAll);
   const { data, isSuccess, isPending } = trpc.viewer.insights.averageEventDuration.useQuery(
     {
-      startDate: startDate.toISOString(),
-      endDate: endDate.toISOString(),
+      startDate: dayjs(startDate).utc(true).toISOString(),
```
Comment (emrysal) [inline packages/features/insights/components/AverageEventDurationChart.tsx:22]: I don't think going .utc(true) is a good idea here - utc() is better suited. it comes in (startDate) as UTC too, however you could simplify then by going dayjs.utc(startDate).

**Review by emrysal — DISMISSED** [review]
Minor nits, found a possible bug - but approved as the bug won't matter much

**Review by keithwillcode — DISMISSED** [review]

**Review by emrysal — APPROVED** [review]

**Pull request merged**